src/components/DateRangePicker.jsx

```diff
@@ -187,10 +188,15 @@ class DateRangePicker extends React.Component {
   }
 
   onDateRangePickerInputFocus(focusedInput) {
-    const { onFocusChange, withPortal, withFullScreenPortal } = this.props;
+    const {
+      onFocusChange,
+      withPortal,
+      withFullScreenPortal,
+      keepFocusOnInput
+    } = this.props;
 
     if (focusedInput) {
-      const moveFocusToDayPicker = withPortal || withFullScreenPortal || this.isTouchDevice;
+      const moveFocusToDayPicker = !keepFocusOnInput && (withPortal || withFullScreenPortal || this.isTouchDevice);
```

Comment:
One thing that's of note here is that means it's possible to keep focus on the input when using the portal options (with which the inputs are explicitly not visible). 

Maybe we want this to be `(withPortal || withFullScreenPortal || (!keepFocusOnInput && this.isTouchDevice))`? I'm not sure.